git_creation_and_tracking/git_creation_and_tracking.md

```diff
@@ -0,0 +1,780 @@
+[( )] "add a line that the Martian atmosphere is only 1% oxygen"
+[(X)] "add comments on Martian atmosphere"
+***
+<div class = "answer">
+The commit message should describe what this commit does. If you want to know the line by line changes you should use the `diff` command. The purpose of the message is so that you know what you changed without having to read all the individual changes, which is especially helpful if you edited many lines or files.
+</div>
+***
+
+## Understanding the Git workflow
+
+There were a lot of steps to getting our changes saved in Git! You could memorize that sequence of steps, and that is what [many people do](https://xkcd.com/1597/). However you might remember them better if you understand what each is doing.
+
+
+So far we have encountered files that in three states:
+
+- **Working Directory** The working directory refers to the files you are currently working on, regardless of whether Git is tracking them. When you first create a new file, it exists solely in your working directory.
+
+- **Staged Files** When you use the `git add file.txt` command to tell Git to keep track of your file, the file is now staged. Git is paying attention to the file, and knows what is in it, but has not yet created a record of its current state.
+
+
+- **Committed Files** When you commit using the `git commit -m "short description of changes"` command, every file that had been staged is now committed. In other words, a record of the file is saved and you will be able to come back and see that exact version later on if you wish.
+
+As you work on your project, files will cycle through these different states. When you make a change to a file, that change is in your working directory until you restart the `add` and `commit` cycle.
+
+
+**Is there a better way to think of this?** Lots of people have favorite metaphors for how Git works. Before checking out the selection on the next page, do you have any ideas for possible ways to understand the `add` and `commit` process in Git?
+
+### A few metaphors for Git
+Metaphors for Git abound, and no metaphor has been great enough to become the "standard" example. If you are looking for a good enough metaphor to help you understand what Git is, here are some to consider:
+
+* A [poetry metaphor](https://www.youtube.com/watch?v=BCQHnlnPusY) video on YouTube ~14 minutes
+
+* A [musical metaphor](https://www.youtube.com/watch?v=S9Do2p4PwtE) video on YouTube ~5 minutes
+
+* A [metaphor about building a house](https://towardsdatascience.com/a-simple-story-to-explain-version-control-to-anyone-5ab4197cebbc) blog post
+
+* A [metaphor about photo albums](https://www.freecodecamp.org/news/git-the-laymans-guide-to-understanding-the-core-concepts/) article
+
+Remember a metaphor doesn't need to be perfect to be helpful. If you find or come up with one that helps you understand the workflow, use it!
+
+### Telling Git not to track some files
+
+What if we have files that we do not want Git to track for us,
```

Comment:
This section has some line breaks in the middle of sentences... I don't see that showing up in the rendered version, but maybe let text wrap?  Unsure of whether this is some sort of weird github artifact that isn't real.

Reply:
I think it is an artifact from SWCarpentry, fixing it everywhere I see it!